polars/polars-lazy/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -135,7 +167,25 @@ pub(super) fn temporal_range_dispatch(
             }
             builder.finish().into_series()
         }
-        DataType::Datetime(_, _) | DataType::Time => {
+        DataType::Datetime(tu, ref tz) => {
```

Comment:
separating the `Datetime` and `Time` arms to make use of `tu` and `tz` from `Datetime`